apps/openchallenges/infra/cdk/README.md

```diff
@@ -154,24 +227,150 @@ The bucket stack exports the following values:
 
 These outputs can be referenced by other stacks or services.
 
-## Testing
+## Phase 1: Networking and Load Balancer
 
-### Run All Tests
+This phase deploys the core networking infrastructure and application load balancer.
 
-```bash
-nx test openchallenges-infra-cdk
-```
+### Resources Deployed
 
-### Run Unit Tests Only
+#### VPC (Virtual Private Cloud)
+
+- **Configuration**:
+
+  - 2 Availability Zones for redundancy
+  - CIDR block: 10.0.0.0/16 (configurable via `VPC_CIDR` environment variable)
+  - Public subnets (/24): For ALB and NAT Gateway(s)
+  - Private subnets (/24): For ECS tasks and backend services
+  - DNS support: Enabled
+
+- **NAT Gateway Configuration** (Environment-Specific):
+
+  - **Development**: 1 NAT Gateway
+
+    - Cost-optimized: ~$32.50/month
+    - Trade-off: Single point of failure for internet access from private subnets
+    - Suitable for dev/test environments where cost is prioritized over uptime
+
+  - **Stage/Production**: NAT Gateways = Number of Availability Zones (default: 2)
+    - High availability: ~$32.50/month per NAT Gateway
+    - One NAT Gateway per AZ ensures redundancy
+    - If one NAT or AZ fails, the other continues to provide internet access
+    - Recommended for production workloads requiring high uptime
+    - Configurable via `MAX_AZS` environment variable (scales automatically)
+
+#### Application Load Balancer (ALB)
+
+- **Configuration**:
+
+  - Internet-facing: Accessible from the public internet
+  - HTTP support: Always enabled on port 80
+  - HTTPS support: Optional, enabled when `CERTIFICATE_ARN` is provided
+  - Security groups: Allow HTTP (80) and HTTPS (443) from anywhere (0.0.0.0/0)
+  - Health check endpoint: `/health` returns 200 OK with JSON response
+  - Default action: Returns 503 "Service Unavailable" for unconfigured routes
+  - HTTP to HTTPS redirect: Automatically enabled when certificate is configured
+
+- **Listener Rules**:
+  - Priority 1: `/health` → Fixed response (200 OK, JSON)
+  - Default: All other paths → 503 Service Unavailable (until backends are configured)
+
+### CloudFormation Outputs
+
+#### VPC Stack
+
+- `VpcId`: ID of the VPC
+- `PublicSubnetIds`: Comma-separated list of public subnet IDs
+- `PrivateSubnetIds`: Comma-separated list of private subnet IDs
+
+#### ALB Stack
+
+- `LoadBalancerArn`: ARN of the Application Load Balancer
+- `LoadBalancerDnsName`: DNS name to access the ALB
+- `HealthCheckUrl`: Full URL to the health check endpoint
+
+### Environment Configuration
+
+Add these optional variables to your `.env` files:
```

Comment:
this is just a duplicate of what's already in the example .env files.  instead of maintaining this information in both places why not just link to the files instead?

Note: the example .env files seems to be missing the `MAX_AZS` option.